### ISSUE TITLE
Build isn't including the updated --memory-size flags?

### ISSUE THREAD

**Issue by jgeiger** [zchee/docker-machine-driver-xhyve#40]
I've been updating and building from master, and it seems that the resulting binary doesn't have the correct flags generated. I'm wondering if it's something odd I'm doing or there's just an issue elsewhere.

From the root of the project, I do a make install and when I try to a create a machine, it fails showing the non-memory-size flag.

docker-machine create --driver xhyve --xhyve-cpu-count=8 --xhyve-memory-size=8192

**Comment by zchee** [comment]
@jgeiger 
Machine name does not exist in your comments.
Is Miss comment?

BTW, Could you try 

- `docker-machine create --driver xhyve`
- `docker-machine create --driver xhyve --xhyve-cpu-count=8 --xhyve-memory-size=8192 xhyve-test`

and Could you write the result?

**Comment by zchee** [comment]
@jgeiger Oh, I forgot.
Thanks issue :)

**Comment by jgeiger** [comment]
It's pretty much the default output, but it doesn't recognize the changed flag.

Clearly it's in the source, but when I run make install, it doesn't pick that up. I'm wondering if there's just something wrong with the build, or how I'm doing it.

Thanks.

```
➜  ~  docker-machine create --driver xhyve --xhyve-cpu-count=8 --xhyve-memory-size=8192 default
Incorrect Usage.

Usage: docker-machine create [OPTIONS] [arg...]

Create a machine

Description:
   Run 'docker-machine create --driver name' to include the create flags for that driver in the help text.

Options:

   --driver, -d "none"															Driver to create machine with.
   --engine-env [--engine-env option --engine-env option]										Specify environment variables to set in the engine
   --engine-insecure-registry [--engine-insecure-registry option --engine-insecure-registry option]					Specify insecure registries to allow with the created engine
   --engine-install-url "https://get.docker.com"											Custom URL to use for engine installation [$MACHINE_DOCKER_INSTALL_URL]
   --engine-label [--engine-label option --engine-label option]										Specify labels for the created engine
   --engine-opt [--engine-opt option --engine-opt option]										Specify arbitrary flags to include with the created engine in the form flag=value
   --engine-registry-mirror [--engine-registry-mirror option --engine-registry-mirror option]						Specify registry mirrors to use
   --engine-storage-driver 														Specify a storage driver to use with the engine
   --swarm																Configure Machine with Swarm
   --swarm-addr 															addr to advertise for Swarm (default: detect and use the machine IP)
   --swarm-discovery 															Discovery service to use with Swarm
   --swarm-host "tcp://0.0.0.0:3376"													ip/socket to listen on for Swarm master
   --swarm-image "swarm:latest"														Specify Docker image to use for Swarm [$MACHINE_SWARM_IMAGE]
   --swarm-master															Configure Machine to be a Swarm master
   --swarm-opt [--swarm-opt option --swarm-opt option]											Define arbitrary flags for swarm
   --swarm-strategy "spread"														Define a default scheduling strategy for Swarm
   --tls-san [--tls-san option --tls-san option]											Support extra SANs for TLS certs
   --xhyve-boot-cmd "loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 base host=boot2docker"	Command of booting kexec protocol [$XHYVE_BOOT_CMD]
   --xhyve-boot2docker-url 														The URL of the boot2docker image. Defaults to the latest available version [$XHYVE_BOOT2DOCKER_URL]
   --xhyve-cpu-count "1"														Number of CPUs for the machine (-1 to use the number of CPUs available) [$XHYVE_CPU_COUNT]
   --xhyve-disk-size "20000"														Size of disk for host in MB [$XHYVE_DISK_SIZE]
   --xhyve-experimental-nfs-share													Setup NFS shared folder (requires root) [$XHYVE_EXPERIMENTAL_NFS_SHARE]
   --xhyve-memory "1024"														Size of memory for host in MB [$XHYVE_MEMORY_SIZE]
flag provided but not defined: -xhyve-memory-size
```

**Comment by zchee** [comment]
@jgeiger Hmm, I do not know yet the cause...

Could you try this commands?

- `which -a docker-machine-driver-xhyve`
- `which -a docker-machine-xhyve`

**Comment by zchee** [comment]
@jgeiger 
Since the flag is proper, I do not think of only thing that remains is the old binary... :(
https://github.com/zchee/docker-machine-driver-xhyve/blob/master/xhyve.go#L101

**Comment by jgeiger** [comment]
➜  ~  which -a docker-machine-driver-xhyve
/usr/local/bin/docker-machine-driver-xhyve
➜  ~  which -a docker-machine-xhyve
docker-machine-xhyve not found

**Comment by zchee** [comment]
@jgeiger Hmm.

Could you try create test vm with `--debug` flag?

```bash
docker-machine --debug create --driver xhyve xhyve-test
```

It will be output `docker-machine-driver-xhyve` version when the create command.

Latest version: `0.1.0 (66ffe45)`
Latest commit: https://github.com/zchee/docker-machine-driver-xhyve/commit/66ffe45d23ba2bbea6f2eb3524da8a165b6123b7

```bash
Docker Machine Version: 0.5.3-dev, build 5dd64c6
Found binary path at /usr/local/bin/docker-machine-driver-xhyve
Launching plugin server for driver xhyve
Plugin server listening at address 127.0.0.1:61530
() DBG | operation not supported by device
() Calling .GetVersion
Using API Version 1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetCreateFlags
Making call to close connection to plugin binary
Making call to close driver server
(flag-lookup) Calling .Close
Successfully made call to close driver server
Found binary path at /usr/local/bin/docker-machine-driver-xhyve
Launching plugin server for driver xhyve
Plugin server listening at address 127.0.0.1:61534
() DBG | operation not supported by device
() Calling .GetVersion
Using API Version 1
() Calling .SetConfigRaw
() Calling .GetMachineName
(test) Calling .GetMachineName
(test) Calling .DriverName
(test) Calling .GetCreateFlags
(test) Calling .SetConfigFromFlags
Running pre-create checks...
(test) Calling .PreCreateCheck
#              THIS LINE
(test) DBG | ===== Docker Machine xhyve Driver Version 0.1.0 (66ffe45) =====
#              THIS LINE
(test) DBG |
(test) DBG | executing: /usr/local/bin/VBoxManage -v
(test) DBG | STDOUT: 5.0.12r104815
(test) DBG |
(test) DBG | STDERR:
(test) Calling .GetConfigRaw
```

**Comment by zchee** [comment]
@jgeiger If the problem is not resolved, I will prepare for the binary release.

**Comment by jgeiger** [comment]
```
(xhyve-test) DBG | ===== Docker Machine xhyve Driver Version 0.1.0 (HEAD) =====
```
```
git log -10
66ffe45 2015-12-30 13:34:31 +0900 Koichi Shiraishi | Merge pull request #42 from zchee/fix-stop-kill
642b6f0 2015-12-30 13:20:07 +0900 Koichi Shiraishi | Add log infomation
40626ef 2015-12-30 12:38:02 +0900 Koichi Shiraishi | Fix Restart()
299341d 2015-12-30 11:41:43 +0900 Koichi Shiraishi | Fix Kill() to syscall.SIGKILL
c6b983b 2015-12-30 11:41:40 +0900 Koichi Shiraishi | Fix Stop() to syscall.SIGTERM
d76be12 2015-12-30 11:18:58 +0900 Koichi Shiraishi | Merge pull request #41 from zchee/fix-get-vmstate
```

**Comment by zchee** [comment]
@jgeiger GitCommit version `HEAD` is incorrect.

Parse the Git commit during the `make` command, and rewrite the GitCommit version.
https://github.com/zchee/docker-machine-driver-xhyve/blob/master/Makefile#L58
also
https://github.com/zchee/docker-machine-driver-xhyve/blob/master/version.go#L13

Could you try this and write result?

```bash
mv /usr/local/bin/docker-machine-driver-xhyve{,.bak}
# or
rm -f /usr/local/bin/docker-machine-driver-xhyve

cd $GOPATH/src/github.com/zchee/docker-machine-driver-xhyve
make clean
make
make install
```

**Comment by jgeiger** [comment]
It seems building the driver doesn't work the way I'm expecting. If I do the above commands it works, after updating your repo inside my GOPATH.

I think it might be my unfamiliarity with how your build process works (as it seems to be only using the code from inside the GOPATH and looking specifically for your repo)

I'll mess around with it locally and see what I can get working and post back with any solution.

Thanks for the help.

**Comment by zchee** [comment]
@jgeiger Hi,
Sorry to keep you waiting.

I released v0.2.0. Also added pre-compiled binary use travis.
See https://github.com/zchee/docker-machine-driver-xhyve/releases.

If solved of problem use release binary, Could you close issue or comments?

**Comment by jgeiger** [comment]
Hello,

I was able to figure out the build issue. The make file was using your username, so it kept looking at your code in my GOPATH. Once I changed that, I was able to build locally.

Thanks for the release as well.
